v1/bundle/store.go

```diff
@@ -668,44 +693,74 @@ func eraseData(ctx context.Context, store storage.Store, txn storage.Transaction
 	return nil
 }
 
-func erasePolicies(ctx context.Context, store storage.Store, txn storage.Transaction, parserOpts ast.ParserOptions, roots map[string]struct{}) (map[string]*ast.Module, error) {
+func erasePolicies(ctx context.Context, store storage.Store, txn storage.Transaction, parserOpts ast.ParserOptions, roots map[string]struct{}) (map[string]*ast.Module, []string, error) {
 
 	ids, err := store.ListPolicies(ctx, txn)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	getRegoVersion := func(modId string) (ast.RegoVersion, bool) {
+		regoVersionValue, err := store.Read(ctx, txn, moduleRegoVersionPath(modId))
+		if err != nil {
+			return ast.RegoUndefined, false
+		}
+
+		bs, err := json.Marshal(regoVersionValue)
+		if err != nil {
+			return ast.RegoUndefined, false
+		}
+
+		var moduleRegoVersion ast.RegoVersion
+		err = util.UnmarshalJSON(bs, &moduleRegoVersion)
+		if err != nil {
+			return ast.RegoUndefined, false
+		}
+
+		return moduleRegoVersion, true
 	}
 
 	remaining := map[string]*ast.Module{}
+	var removed []string
 
 	for _, id := range ids {
 		bs, err := store.GetPolicy(ctx, txn, id)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
-		module, err := ast.ParseModuleWithOpts(id, string(bs), parserOpts)
+
+		parserOptsCpy := parserOpts
+		if regoVersion, ok := getRegoVersion(id); ok {
```

Comment:
We're doing a store read on every policy id. Is there a performance hit here especially for large bundles? Can we read the entire data once and then use that in the loop? I don't think there would be much performance overhead. Maybe I'm missing something here and we need to structure the data this way. Multiple bundles, maybe?